File: hw/astudillo/U3/Hw19Composite/Composite/src/main/java/ec/edu/espe/composite/composite/model/Supervisor.java
package ec.edu.espe.composite.composite.model;

import java.util.Vector;

public abstract class Supervisor extends Employee {
    protected Vector directReports = new Vector();
    
    @Override
    public void stateName(String title, String name) {
        super.stateName(title, name);
        if (!directReports.isEmpty()) {
            for (int i = 0; i < directReports.size(); ++i) {
                Employee emp = (Employee) directReports.elementAt(i);
                emp.stateName(emp.title, emp.name);
            }
        }
    }
    
    public void add(Employee anEmployee) {
        this.directReports.addElement(anEmployee);
    }
}

File: hw/unda/U3/HW19Composite/CompositePy/model/Supervisor.py
from model.Employee import Employee

class Supervisor(Employee):
    def __init__(self, name=""):
        super().__init__(name)
        self.direct_reports = []

    def add(self, employee):
        self.direct_reports.append(employee)

    def state_name(self):
        super().state_name()
        for emp in self.direct_reports:
            emp.state_name()
